internal/services/netapp/netapp_volume_resource.go

```diff
@@ -323,6 +324,36 @@ func resourceNetAppVolume() *pluginsdk.Resource {
 				},
 			},
 
+			"data_protection_backup_policy": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"backup_policy_id": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: azure.ValidateResourceID,
```

Comment:
We should be able to use the resource specific validation function here instead 
```suggestion
							ValidateFunc: backuppolicy.ValidateBackupPolicyID,
```

Reply:
For some reason I can't do that, when using specific resource validation, it seems that the hashicorp's azure sdk fails to validate the resource id.

```     testcase.go:173: Step 1/3 error: Error running apply: exit status 1

        Error: parsing "/subscriptions/66bc9830-19b6-4987-94d2-0e487be7aa47/resourceGroups/acctestRG-netapp-241018232302838130/providers/Microsoft.NetApp/netAppAccounts/acctest-NetAppAccount-241018232302838130/backupPolicies/acctest-NetAppBackupPolicy-241018232302838130": parsing segment "staticMicrosoftDataProtection": parsing the BackupPolicy ID: the segment at position 5 didn't match

        Expected a BackupPolicy ID that matched:

        > /subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/example-resource-group/providers/Microsoft.DataProtection/backupVaults/backupVaultName/backupPolicies/backupPolicyName

        However this value was provided:

        > /subscriptions/66bc9830-19b6-4987-94d2-0e487be7aa47/resourceGroups/acctestRG-netapp-241018232302838130/providers/Microsoft.NetApp/netAppAccounts/acctest-NetAppAccount-241018232302838130/backupPolicies/acctest-NetAppBackupPolicy-241018232302838130

        The parsed Resource ID was missing a value for the segment at position 5
        (which should be the name of the Resource Provider [for example 'Microsoft.DataProtection']).



          with azurerm_netapp_volume.test,
          on terraform_plugin_test.tf line 144, in resource "azurerm_netapp_volume" "test":
         144:     backup_policy_id = azurerm_netapp_backup_policy.test.id

=== NAME  TestAccNetAppVolume_backupPolicyUpdate
    testcase.go:173: Step 1/6 error: Error running apply: exit status 1

        Error: parsing "/subscriptions/66bc9830-19b6-4987-94d2-0e487be7aa47/resourceGroups/acctestRG-netapp-241018232302838846/providers/Microsoft.NetApp/netAppAccounts/acctest-NetAppAccount-241018232302838846/backupPolicies/acctest-NetAppBackupPolicy-241018232302838846": parsing segment "staticMicrosoftDataProtection": parsing the BackupPolicy ID: the segment at position 5 didn't match

        Expected a BackupPolicy ID that matched:

        > /subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/example-resource-group/providers/Microsoft.DataProtection/backupVaults/backupVaultName/backupPolicies/backupPolicyName

        However this value was provided:

        > /subscriptions/66bc9830-19b6-4987-94d2-0e487be7aa47/resourceGroups/acctestRG-netapp-241018232302838846/providers/Microsoft.NetApp/netAppAccounts/acctest-NetAppAccount-241018232302838846/backupPolicies/acctest-NetAppBackupPolicy-241018232302838846

        The parsed Resource ID was missing a value for the segment at position 5
        (which should be the name of the Resource Provider [for example 'Microsoft.DataProtection']).



          with azurerm_netapp_volume.test,
          on terraform_plugin_test.tf line 144, in resource "azurerm_netapp_volume" "test":
         144:     backup_policy_id = azurerm_netapp_backup_policy.test.id

--- FAIL: TestAccNetAppVolume_backupPolicyUpdate (198.67s)
```

Reply:
Here is the testing result rolling back this change, I'll leave as generic now until sdk is fixed since it looks like it is an issue coming from there.


```
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/netapp -parallel 5 -run=TestAccNetAppVolume_backupPolicy -count=1 -timeout 1200m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccNetAppVolume_backupPolicy
=== PAUSE TestAccNetAppVolume_backupPolicy
=== RUN   TestAccNetAppVolume_backupPolicyUpdate
=== PAUSE TestAccNetAppVolume_backupPolicyUpdate
=== CONT  TestAccNetAppVolume_backupPolicy
=== CONT  TestAccNetAppVolume_backupPolicyUpdate
--- PASS: TestAccNetAppVolume_backupPolicyUpdate (1268.83s)
--- PASS: TestAccNetAppVolume_backupPolicy (1356.14s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp        1356.158s
```